core/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -407,13 +408,23 @@ public CreateIndexRequest source(Map<String, ?> source) {
                     } catch (IOException e) {
                         throw new ElasticsearchParseException("failed to parse custom metadata for [{}]", name);
                     }
+                } else {
+                    // found a key which is neither custom defined nor one of the supported ones
+                    if (unsupportedKey == null) {
```

Comment:
See https://github.com/elastic/elasticsearch/blob/5.x/core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java#L98 for an example